chain/sync.go

```diff
@@ -845,6 +845,14 @@ loop:
 	if err != nil {
 		return nil, xerrors.Errorf("failed to load next local tipset: %w", err)
 	}
+
+	if chkpt := syncer.store.GetCheckpoint(); !ignoreCheckpoint && chkpt != nil {
```

Comment:
Is this not going to be a bit racy? I guess maybe that's ok? Will we try to re-sync otherwise?

In general, I'm not sure why we're not just doing this with heights. I.e., if `known` has a height less than or equal to the checkpoint, don't try to sync that chain?

Reply:
You are correct. That would work great as well.
The majority of the thing here was trying to get the test working.